### PR TITLE
Unbounded knapsack doc

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #171

### CODE REVIEW

**Comment by SAY-droid427** [comment]
@HarshCasper When I create this PR, I don't know why but some of the previous commits made on some other branch which has been merged also get appended to it . But in files changed it shows only the changes that I have made for the unbounded-knapsack problem. Could you please check once if it is okay

**Comment by HarshCasper** [comment]
> @HarshCasper When I create this PR, I don't know why but some of the previous commits made on some other branch which has been merged also get appended to it . But in files changed it shows only the changes that I have made for the unbounded-knapsack problem. Could you please check once if it is okay

Looks good to me!

File: docs/dynamic-programming/unbounded-knapsack.mdx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+#### Output
+
+```
+Value : 20
+```
+
+## Complexity analysis
+
+Time Complexity : `O(n*W)`
+Space Complexity : `O(n*W)`
```
Comment (HarshCasper) [inline docs/dynamic-programming/unbounded-knapsack.mdx:203]: Put them as bullet points

File: docs/dynamic-programming/unbounded-knapsack.mdx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        {
+            if (i == 0 || j == 0)
+                t[i][j] = 0;
+        }
+    }
+
+    /*-->If the weight of the ith item is less than the weight of the knapsack, then we can either include the item or exclude it depending upon which will maximise the value
+      -->If included then we can include it again
+      -->If not included once then it is not included again
+      -->If the weight is greater than the weight of the knapsack then we cannot include it into the knapsack at any time
```
Comment (HarshCasper) [inline docs/dynamic-programming/unbounded-knapsack.mdx:84]: Don't use `-->`.  Keep them as ordered list

File: docs/dynamic-programming/unbounded-knapsack.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+slug: /dynamic-programming/unbounded-knapsack
+---
+
+## Introduction
+
+Given a knapsack of weight W and n number of items, we need to calculate the maximum value of the items that can be packed into the knapsack such that we can choose any item any number of times. Here we assume that each item has an infinite supply.
+
+## Explanation
+
+ We use dynamic programming to solve this problem. We start from the nth item. There are two possibilities at each step: the weight of the item is less than the knapsack W or greater than the weight of the knapsack W. If the weight of the item is greater than the weight of the knapsack W, then we cannot include the item in it. But if the weight is less than the weight of the knapsack, then we again have two choices to maximize the value of the knapsack: we either choose it or not. If we donot choose it, then the next time we select from n-1 items available to us. If we choose the nth item, then next time we can choose from n items again. So there is a problem of smaller sub problems and so dynamic programming can be used to solve the problem.
```
Comment (HarshCasper) [inline docs/dynamic-programming/unbounded-knapsack.mdx:12]: Use backticks around values

File: docs/dynamic-programming/unbounded-knapsack.mdx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
```
Comment (HarshCasper) [inline docs/dynamic-programming/unbounded-knapsack.mdx:13]: Adding diagrams would be preferred here

File: docs/dynamic-programming/unbounded-knapsack.mdx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+#### Example 1
+
+```
+Let capacity W be `50` and  number of items be `2`.
+The weights are `5, 10`.
+The values are  `50, 100`.
+We start from the nth item whose weight is 10 which is less than W. So we can either choose it or not depending on what will maximise the value of the knapsack.
+We get greater value when including the nth item than when we get by excluding it.
+The maximum value comes out to be `5*100=500`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/unbounded-knapsack.mdx:24]: Backticks don't work here

File: docs/dynamic-programming/unbounded-knapsack.mdx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+```
+#### Example 2
+
+```
+Let capacity W be `5` and number of items be `2`.
+The weights are `3,5`.
+The values of items are `10,15`.
+We start from the nth item whose weight is equal to the knapsack W, so we may either include it or exclude it. If we include it the value of the knapsack will be `15`, but if we exclude it the value of the knapsack will be `10`. 
+So the maximum value of the knapsack will be `15`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/unbounded-knapsack.mdx:33]: Backticks don't work here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/dynamic-programming/unbounded-knapsack.mdx (hunk 1 above)
Comment (SAY-droid427) [reply docs/dynamic-programming/unbounded-knapsack.mdx:203]: Done

File: docs/dynamic-programming/unbounded-knapsack.mdx (hunk 2 above)
Comment (SAY-droid427) [reply docs/dynamic-programming/unbounded-knapsack.mdx]: Changed them to ordered list

File: docs/dynamic-programming/unbounded-knapsack.mdx (hunk 3 above)
Comment (SAY-droid427) [reply docs/dynamic-programming/unbounded-knapsack.mdx:12]: Removed backticks

File: docs/dynamic-programming/unbounded-knapsack.mdx (hunk 4 above)
Comment (SAY-droid427) [reply docs/dynamic-programming/unbounded-knapsack.mdx:13]: I think it is difficult to explain through diagram in this case.

File: docs/dynamic-programming/unbounded-knapsack.mdx (hunk 5 above)
Comment (SAY-droid427) [reply docs/dynamic-programming/unbounded-knapsack.mdx]: Removed

File: docs/dynamic-programming/unbounded-knapsack.mdx (hunk 6 above)
Comment (SAY-droid427) [reply docs/dynamic-programming/unbounded-knapsack.mdx]: Removed

**Comment by SAY-droid427** [comment]
@HarshCasper I checked the details of the check failing, it shows due to "golomb sequence". Image attached. Checks were passing before I removed the merge conflict and merged main into my branch.
ERRO

**Comment by SAY-droid427** [comment]
@HarshCasper  Please review this PR. I have made all the changes asked.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**